deep_qa/layers/adaptive_recurrence.py

```diff
@@ -65,7 +66,7 @@ class AdaptiveStep(MaskedLayer):
     probabilities, states and outputs from a timestep t's contribution if they have already reached
     1 - epsilon at a timestep s < t.
     '''
-    def __init__(self, memory_network, params: "Params",
+    def __init__(self, memory_network, params,
```

Comment:
Instead of having params here, you should instead have `epsilon`, `max_computation`, etc., as named arguments to `__init__`, just like `initialization` and `name` are.  Though, `name` doesn't really matter here, because it gets picked up by the super class already.  If you notice the `TODO` down in `get_config`, it's because of exactly this issue.  The keys there need to match the named arguments to `__init__`, in order to conform to the Keras layer API.